### PR TITLE
Migrate Figure.subplot tests to dvc

### PR DESCRIPTION
**Pull request by michaelgrund**
**Description of proposed changes**

Related to #1131.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/tests/test_subplot.py [hunk 1]
```diff
@@ -23,66 +22,63 @@ def test_subplot_basic_frame():
...
 
-    with fig_test.subplot(
-        autolabel=True, margins=["0.3c", "0.1c"], title="Subplot Title", **kwargs
+    with fig.subplot(
+        autolabel=True,
+        margins=["0.3c", "0.1c"],
+        title="Subplot Title",
+        nrows=2,
+        ncols=1,
+        figsize=("15c", "6c"),
```
Comment (weiji14) [inline pygmt/tests/test_subplot.py:51]: Put the required arguments (nrows/ncols/figsize) first, before the optional arguments (autolabel/margins/title).
```suggestion
        nrows=2,
        ncols=1,
        figsize=("15c", "6c"),
        autolabel=True,
        margins=["0.3c", "0.1c"],
        title="Subplot Title",
```

File: pygmt/tests/test_subplot.py [hunk 2]
```diff
@@ -23,66 +22,63 @@ def test_subplot_basic_frame():
...
-    with fig_test.subplot(
-        clearance=["s0.2c", "n0.2c"], sharex="t", sharey=True, **kwargs
+    with fig.subplot(
+        clearance=["s0.2c", "n0.2c"],
+        sharex="t",
+        sharey=True,
+        nrows=2,
+        ncols=2,
+        frame="WSrt",
+        figsize=("5c", "5c"),
```
Comment (weiji14) [inline pygmt/tests/test_subplot.py:74]:
```suggestion
        nrows=2,
        ncols=2,
        figsize=("5c", "5c"),
        frame="WSrt",
        clearance=["s0.2c", "n0.2c"],
        sharex="t",
        sharey=True,
```

**Comment by weiji14** [comment]
Btw, we should teach you about making batch changes to save on a few clicks. See https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes. You can go to the 'Files' tab on the GitHub UI and click on 'Add suggestion to batch'.

image

**Comment by michaelgrund** [comment]
> 
> 
> Btw, we should teach you about making batch changes to save on a few clicks. See https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes. You can go to the 'Files' tab on the GitHub UI and click on 'Add suggestion to batch'.
> 
> image

Thanks for mentioning @weiji14, didn't know that so far 😄.

**Review by weiji14 — APPROVED** [review]
Looks good @michaelgrund, feel free to merge this in yourself. I've disabled the 'Docs' CI test requirement (since they don't run on changes to `pygmt/tests/*') so you should be able to merge it in straight this time.

**Pull request merged**